fairseq/model_parallel/models/pipeline_parallel_transformer/model.py

```diff
@@ -39,15 +39,38 @@
 
 DEFAULT_MAX_SOURCE_POSITIONS = 1024
 DEFAULT_MAX_TARGET_POSITIONS = 1024
+TORCH_PIPE = False
+try:
+    from torch.distributed.pipeline.sync import Pipe
+    from torch.distributed.pipeline.sync.utils import partition_model
+    from torch.distributed import rpc
+    import tempfile
+    TORCH_PIPE = True
+    # Initialize single process RPC agent since TORCH_PIPE requires
+    # RRef.
+    tmpfile = tempfile.NamedTemporaryFile()
+    rpc.init_rpc(
+        name="worker",
+        rank=0,
+        world_size=1,
+        rpc_backend_options=rpc.TensorPipeRpcBackendOptions(
+            init_method="file://{}".format(tmpfile.name),
+        )
+    )
```

Comment:
Can you add some more details on why Torch Pipe needs this rpc.init_rpc() call?

Reply:
Updated the comment on line 49 for more information. Basically, the pipe returns an RRef which depends on RPC being initialized.